src/utils/utils.ts

```diff
@@ -117,8 +121,16 @@ export const getAPIUri = async (identifierType: string, identifier: string) => {
  * @param {string} apiUrl - the url to make a request to
  */
 
-export const apiCall = async (apiUrl: string) => {
+export const apiCall = async (
+  apiUrl: string,
+  urlParam?: { [key: string]: any }
```

Comment:
I added an optional `urlParam` so that that we could add the needed url key/param to the query string when calling the `featured` endpoint.  If the `random` param is not set to `true` in the query string, then numerous results are returned per [code below](https://github.com/NYPL/repository-api/blob/qa/app/controllers/api/v2/items_controller.rb#L105):
<img width="959" alt="Screenshot 2024-03-08 at 7 29 07 AM" src="https://github.com/NYPL/digital-collections/assets/4614468/93e7eb09-38ee-434d-af1a-2da2799410a1">

**Example:**
<img width="977" alt="Screenshot 2024-03-08 at 7 32 20 AM" src="https://github.com/NYPL/digital-collections/assets/4614468/4b4c1e99-34cb-4deb-866c-4401166dbffe">

Reply:
Good point. I think we'll only call the endpoint with the `?random=true` query param so this code should expect only one result. This is good to know if this changes in the future.